libs/designer-ui/src/lib/tokenpicker/tokenpickersection/tokenpickeroption.tsx

```diff
@@ -177,42 +188,43 @@ export const TokenPickerOptions = ({
-            <span> {section.label}</span>
-            {searchQuery || !hasAdvanced(section.tokens) ? null : (
+            <span>{section.label}</span>
+            {!showSeeMoreOrLessButton ? null : (
               <button
                 className="msla-token-picker-section-header-button"
                 onClick={handleMoreLess}
                 data-automation-id={`msla-token-picker-section-header-button-${convertUIElementNameToAutomationId(section.label)}`}
               >
-                <span> {moreOptions ? buttonTextMore : buttonTextLess}</span>
+                <span>{moreOptions ? buttonTextMore : buttonTextLess}</span>
               </button>
             )}
           </div>
           <div className="msla-token-picker-section-options">
             {(!searchQuery ? section.tokens : filteredTokens).map((token, j) => {
-              if (!token.isAdvanced || !moreOptions || searchQuery) {
-                return (
-                  <button
-                    className="msla-token-picker-section-option"
-                    data-automation-id={`msla-token-picker-section-option-${j}`}
-                    key={`token-picker-option-${j}`}
-                    onClick={() => handleTokenClicked(token)}
-                  >
-                    <div className="msla-token-picker-section-option-text">
-                      <div className="msla-token-picker-option-inner">
-                        <div className="msla-token-picker-option-title">{token.title}</div>
-                        <div className="msla-token-picker-option-description" title={token.description}>
-                          {token.description}
-                        </div>
+              if ((token.isAdvanced || j >= maxRowsShown) && moreOptions && !searchQuery) {
+                return null;
+              }
+
+              return (
+                <button
+                  className="msla-token-picker-section-option"
+                  data-automation-id={`msla-token-picker-section-option-${j}`}
+                  key={`token-picker-option-${j}`}
+                  onClick={() => handleTokenClicked(token)}
+                >
+                  <div className="msla-token-picker-section-option-text">
+                    <div className="msla-token-picker-option-inner">
+                      <div className="msla-token-picker-option-title">{token.title}</div>
+                      <div className="msla-token-picker-option-description" title={token.description}>
+                        {token.description}
                       </div>
-                      {/* {token.type && token.outputInfo?.type !== TokenType.FX ? (
-                        <div className="msla-token-picker-option-type">{token.type}</div>
-                      ) : null} */}
                     </div>
-                  </button>
-                );
-              }
-              return null;
+                    {/* {token.type && token.outputInfo?.type !== TokenType.FX ? (
```

Comment:
remove?

Reply:
Sure, happy to. Wasn't sure if there was a reason it was kept.